### PR TITLE
Add Static Website example (AWS, TypeScript)

### PR DESCRIPTION
**Pull request by chrsmith**
Example showing off how to host a static website on AWS using Pulumi and TypeScript.

There isn't anything too fancy going on, although the example is a bit more involved because it requires you to manually set up the domain on Route53 and create the certificate on ACM first.

### CODE REVIEW

File: aws-ts-static-website/index.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    certificateArn: stackConfig.require("certificateArn"),
+};
+
+// contentBucket is the S3 bucket that the website's contents will be stored in.
+const contentBucket = new aws.s3.Bucket(
+    "contentBucket",
+    {
+        bucket: config.targetDomain,
+        acl: "public-read",
+        // Configure S3 serve bucket contents as a website. This way S3 will automatically convert
```
Comment (lukehoban) [inline aws-ts-static-website/index.ts]: *to* serve?

File: aws-ts-static-website/index.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+// Sync the contents of the source directory with the S3 bucket, which will in-turn show up on the CDN.
+const webContentsRootPath = path.join(process.cwd(), config.pathToWebsiteContents);
+console.log("Syncing contents from local disk at", webContentsRootPath);
+crawlDirectory(
+    webContentsRootPath,
+    (filePath: string) => {
+        const relativeFilePath = filePath.replace(webContentsRootPath + "/", "");
+        // Create the aws.S3.BucketObject for every file. This has Pulumi track every file change for each
+        // update to the website. However, for large website it may be more efficient to not manage individual
+        // files using pulumi and instead just use the AWS CLI to sync files instead. (e.g. `aws s3 sync`.)
```
Comment (lukehoban) [inline aws-ts-static-website/index.ts]: Perhaps move this comment into the README?

File: aws-ts-static-website/index.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    {
+        bucket: config.targetDomain,
+        acl: "public-read",
+        // Configure S3 serve bucket contents as a website. This way S3 will automatically convert
+        // requests for "foo/" to "foo/index.html".
+        website: {
+            indexDocument: "index.html",
+            errorDocument: "404.html",
+        }
+    });
```
Comment (lukehoban) [inline aws-ts-static-website/index.ts]: Not a big deal, but the formatting of these calls is different than we use in all other examples in this repo (and elsewhere).

```typescript
const contentBucket = new aws.s3.Bucket("contentBucket", {
    bucket: config.targetDomain,
    acl: "public-read",
    // Configure S3 serve bucket contents as a website. This way S3 will automatically convert
    // requests for "foo/" to "foo/index.html".
    website: {
        indexDocument: "index.html",
        errorDocument: "404.html",
     }
});
```

File: aws-ts-static-website/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+- [Amazon CloudFront](https://aws.amazon.com/cloudfront/) is the CDN serving content.
+- [Amazon Route53](https://aws.amazon.com/route53/) is used to set up the DNS for the website.
+- [Amazon Certificate Manager](https://aws.amazon.com/certificate-manager/) is used for securing things via HTTPS.
+
+## Getting Started
+
+Install prerequisites and build the Pulumi program with:
+
+```bash
+yarn install
```
[the remark this answers is not in the document]
Comment (chrsmith) [reply aws-ts-static-website/README.md]: Done.

File: aws-ts-static-website/index.ts (hunk 1 above)
Comment (chrsmith) [reply aws-ts-static-website/index.ts]: Done.

File: aws-ts-static-website/index.ts (hunk 2 above)
Comment (chrsmith) [reply aws-ts-static-website/index.ts]: Done.

File: aws-ts-static-website/index.ts (hunk 3 above)
Comment (chrsmith) [reply aws-ts-static-website/index.ts]: Done. I missed this in the PR, so I'll just make the commit directly into `master`, like some kind of animal.

**Pull request merged**